### PR TITLE
[WIP]: Refactor tm protocol

### PR DESCRIPTION
**Pull request by Adamantios**
## Proposed changes

Refactor tendermint protocol and handler (which are in fact for ACN communication), in order to add support for sharing the recovery parameters.

## Fixes

n/a

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [x] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

n/a

### CODE REVIEW

File: packages/valory/skills/abstract_round_abci/handlers.py [hunk 1]
```diff
@@ -681,12 +715,37 @@ def _handle_response(
...
+        """Process params-sharing response messages."""
+
+        if not self._check_registered(message, dialogue):
+            return
+
+        try:
+            recovery_params = json.loads(message.params)
+            shared_state = cast(SharedState, self.context.state)
+            shared_state.tm_recovery_params = TendermintRecoveryParams(
+                **recovery_params
```
Comment (0xArdi) [inline packages/valory/skills/abstract_round_abci/handlers.py:733]: don't we care about the sender?

Comment (Adamantios) [reply packages/valory/skills/abstract_round_abci/handlers.py:733]: Correct, though I will include this on another PR, which will store all the responses on the shared state and will also have functionality for choosing the most common one.

**Review by 0xArdi — APPROVED** [review]

File: packages/packages.json [hunk 2]
```diff
@@ -4,43 +4,43 @@
...
-        "connection/valory/ledger/0.19.0": "bafybeiadc25se7dgnn4mufztwpzdono4xsfs45qknzdqyi3gckn6ccuv44",
-        "connection/valory/p2p_libp2p_client/0.1.0": "bafybeidkk33xbga54szmitk6uwsi3ef56hbbdbuasltqtiyki34hgfpnxa"
+        "protocol/open_aea/signing/1.0.0": "bafybeihgwrowgtegpe4ixe3iy5jx65u7pxebl5btrkywzmx7g52gjtbnpi",
+        "protocol/valory/acn/1.1.0": "bafybeieldogmtf3m4jdsvt4vvyay3jh54rjn3deasymfw43vz3o42vigmq",
+        "protocol/valory/http/1.0.0": "bafybeifru3qs6udfzprax7jxktbsuzn7immfvi3scgfspifq3zdxwkgvnm",
+        "protocol/valory/ledger_api/1.0.0": "bafybeigqr4y3ykz3iulrcoqmji7hy3dxaoy7zmyyzff4ivpbubcpwdknai",
+        "protocol/valory/contract_api/1.0.0": "bafybeif32nchkgn6yet7e5gt4auhf7lsahxnj4t36kxbw55p3gi7qpeuxq",
+        "connection/valory/http_client/0.23.0": "bafybeigggahci7hq6tr3tyueatgkvgn73y4b3av2vk7vtr7jkeuwsqcteq",
+        "connection/valory/ledger/0.19.0": "bafybeicpxyoxez7lperltamvikxu6vzk2lhqakbivce4nzywyzoqbxoogm",
+        "connection/valory/p2p_libp2p_client/0.1.0": "bafybeiayoofxmj6z3pasn2akqj3udgq2ta2ar6mv6zoehstul2btvv3gqa"
```
Comment (DavidMinarsch) [inline packages/packages.json:44]: This looks suspicious - they should not have changed

File: packages/valory/protocols/tendermint/README.md [hunk 3]
```diff
@@ -16,10 +16,14 @@ description: A protocol for communication between two AEAs to share tendermint c
...
-    query: pt:optional[pt:str]  # required field, optional provides default value, not used otherwise.
-  response:
+  request_genesis_info: 
+    query: pt:optional[pt:str]
+  request_recovery_params:
+    query: pt:optional[pt:str]
+  response_genesis_info:
     info: pt:str
+  response_recovery_params:
+    params: pt:str
```
Comment (DavidMinarsch) [inline packages/valory/protocols/tendermint/README.md:26]: I suggest changing to:

get_genesis_info, genesis_info

o/w there is always this bloat in the performatives

File: packages/valory/skills/abstract_round_abci/handlers.py (hunk 1 above)
Comment (Adamantios) [reply packages/valory/skills/abstract_round_abci/handlers.py:733]: https://github.com/valory-xyz/open-autonomy/pull/1736#discussion_r1067309643

File: packages/valory/protocols/tendermint/README.md (hunk 3 above)
Comment (Adamantios) [reply packages/valory/protocols/tendermint/README.md:26]: I will address this after merging #1736.

Comment (Adamantios) [reply packages/valory/protocols/tendermint/README.md]: Done in e4555a5517fd2ea778295d7ed4105d1cde85ddd6.

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**